src/state/request/tests/reducer.test.js

```diff
@@ -0,0 +1,125 @@
+import deepfreeze from 'deepfreeze';
+
+import reducer from '../reducer';
+import {
+  REQUEST_SET_METHOD,
+  REQUEST_SELECT_ENDPOINT,
+  REQUEST_UPDATE_URL,
+  REQUEST_UPDATE_PATH_PART_VALUE,
+  REQUEST_SET_QUERY_PARAM,
+  REQUEST_SET_BODY_PARAM
+} from '../../actions';
+
+const endpoint = { path_labeled: '/$site/posts' };
+const state = {
+  endpoint,
+  method: 'GET',
+  queryParams: { context: 'view' },
+  bodyParams: { a: 'b' },
+  pathValues: { $site: 'mySite' },
+  url: '/help'
+};
+deepfreeze(state);
```

Comment:
Why not invoke `deepfreeze` as part of the assignment, i.e. `const state = deepfreeze({ /* ... */ });`